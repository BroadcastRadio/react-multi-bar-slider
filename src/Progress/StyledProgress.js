import PropTypes from 'prop-types';
import glamorous from 'glamorous';
import transition from '../utils/transition';
import getHalf from '../utils/getHalf';

const StyledProgress = glamorous.div(
  {
    position: 'absolute',
    top: 0,
  },
  ({
    color,
    progress,
    height,
    equal,
    equalColor,
    roundedCorners,
    reversed,
    noTransition,
    zIndex
  }) => ({
    left: reversed ? 'auto' : 0,
    right: reversed ? 0 : 'auto',
    width: `${progress || 0}%`,
    height,
    backgroundColor: equal && equalColor ? equalColor : color,
    borderRadius: roundedCorners ? getHalf(height) : 0,
    zIndex,
    transition: noTransition ? 'none' : transition
  })
);

StyledProgress.propTypes = {
  color: PropTypes.string.isRequired,
  progress: PropTypes.number.isRequired,
  height: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]).isRequired,
  equal: PropTypes.bool.isRequired,
  equalColor: PropTypes.string,
  roundedCorners: PropTypes.bool.isRequired,
  reversed: PropTypes.bool.isRequired,
  noTransition: PropTypes.bool.isRequired,
  zIndex: PropTypes.number.isRequired,
  css: PropTypes.object
};

StyledProgress.defaultProps = {
  equal: false
};

export default StyledProgress;
