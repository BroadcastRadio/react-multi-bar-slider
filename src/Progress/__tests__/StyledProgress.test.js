import React from 'react';
import Enzyme, { render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import StyledProgress from '../StyledProgress';

Enzyme.configure({ adapter: new Adapter() });

beforeAll(() => {
  global.styledProgressProps = {
    color: 'green',
    progress: 7,
    height: 14,
    equal: true,
    equalColor: 'purple',
    roundedCorners: false,
    reversed: false,
    noTransition: false,
    zIndex: 4
  };
});

afterAll(() => {
  delete global.styledProgressProps;
});

/* global styledProgressProps */
describe('StyledProgress.js', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(<StyledProgress {...styledProgressProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const styledProgress = render(<StyledProgress {...styledProgressProps} />);
    expect(styledProgress).toBeDefined();
  });
});
