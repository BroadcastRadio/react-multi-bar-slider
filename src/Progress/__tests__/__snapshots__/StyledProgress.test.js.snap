// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledProgress.js matches the snapshot 1`] = `
<div
  className="css-yalmu7"
  color="green"
  height={14}
  reversed={false}
/>
`;
